wicket-core/src/main/java/org/apache/wicket/page/PageAccessSynchronizer.java

```diff
@@ -93,8 +94,8 @@ public Duration getTimeout(int pageId)
 	 */
 	public void lockPage(int pageId) throws CouldNotLockPageException
 	{
-		final Thread thread = Thread.currentThread();
-		final PageLock lock = new PageLock(pageId, thread);
+		final RequestCycle cycle = RequestCycle.get();

```

Comment:
PageLock is the lock: as long as there is a PageLock instance, all access to the corresponding page has to wait.

Currently PageLock holds a reference to the thread, but the thread isn't used actually apart from identifying the lock. The RequestCycle can be used equally well for that.